pkg/kncloudevents/message_sender.go

```diff
@@ -160,6 +160,47 @@ func RetryConfigFromDeliverySpec(spec duckv1.DeliverySpec) (RetryConfig, error)
 	return retryConfig, nil
 }
 
-func checkRetry(_ context.Context, resp *nethttp.Response, err error) (bool, error) {
-	return !(resp != nil && resp.StatusCode < 300), err
+// Function to Determine Whether To Retry Based On Response
+//
+// Note - Returning true indicates a retry should occur.  Returning an error will result in that
+//        error being returned instead of any errors from the Request.
+//
+func checkRetry(_ context.Context, response *http.Response, err error) (bool, error) {
```

Comment:
Replacing the original function is a breaking change, I'm not sure we should do that... What about naming the old function `RetryAllGreaterThan300`, the new one [insert_a_fancy_name_here] and then make sure both are public? So one can pick which one wants to use

Reply:
bikeshedding, `RetryIfGreaterThan300` ?

Reply:
yeah, that's true . let me fiddle w/ that